docs/openapi/schemas.yaml

```diff
@@ -1517,3 +1520,91 @@ ImportAssistantResponse:
     choices:
       - message:
           content: '.breadcrumbs'
+ApiKeyRequest:
+  type: object
+  required:
+    - name
+    - feature
+    - domains
+  properties:
+    name:
+      description: The name of the API key
+      type: string
+    feature:
+      description: The feature the API key is for
+      type: string
+    domains:
+      description: The domains the API key is allowed to access
+      type: array
+      items:
+        type: string
+  example:
+    name: 'Example API Key'
+    feature: 'imports'
+    domains:
+      - 'example.com'
+ApiKeyResponse:
+  type: object
+  properties:
+    id:
+      description: The ID of the API key
+      $ref: '#/Id'
+    name:
+      description: The name of the API key
+      type: string
+    apiKey:
+      description: The API key
+      type: string
+    scopes:
+      description: The scopes of the API key
+      type: array
+      items:
+        type: string
```

Comment:
Will this be an `object` type, as they are currently stored in the DB?

Reply:
They will be stored as an array of strings. Not as a map object as we are moving domains to a higher level.

Reply:
Will this make it trickier (more error prone) to serialize from the api-keys table to this response DTO?

I totally see the benefit of the higher level `features` prop and specifying domains at the top level of the "create key" request, but wonder if that should be considered a convenience helper that we don't need to mirror when we actually respond back with the key. After all we will be losing the `features` property anyways, in favour of specifying the exact scopes that a particular feature (or set of features) maps to.

Reply:
I was thinking of modifying the ApiKeys table as well to make it more simplistic. Keep domains at a higher level. I'm open to re-think it.

Reply:
As discussed at standup, let's mirror how we've modelled scopes in the api-keys table.